tests/integration/docidcache/test_crud_cache.py

```diff
@@ -0,0 +1,215 @@
+        else:
+            assert d.text == f'hello world {i}'
+            assert d.embedding.shape == d_embedding.shape
+
+        assert len(d.chunks) == chunks
+
+        for j, c in enumerate(d.chunks):
+            id_int = int(c.id)
+            assert id_int not in ids_used
+            ids_used.add(id_int)
+            if same_content:
+                if chunk_content is None:
+                    chunk_content = c.content_hash
+                assert c.content_hash == chunk_content
+                assert c.text == 'hello world from chunk'
+                np.testing.assert_almost_equal(c.embedding, c_embedding)
+            else:
+                assert c.text == f'hello world from chunk {j}'
+                assert c.embedding.shape == c_embedding.shape
+
+
+@pytest.mark.parametrize('indexers, field, shards, chunks, same_content',
+                         [
+                             ('sequential', 'id', 1, 5, False),
+                             ('sequential', 'id', 3, 5, False),
+                             ('sequential', 'id', 3, 5, True),
+                             ('sequential', 'content_hash', 1, 0, False),
+                             ('sequential', 'content_hash', 1, 0, True),
+                             ('sequential', 'content_hash', 1, 5, False),
+                             ('sequential', 'content_hash', 1, 5, True),
+                             ('sequential', 'content_hash', 3, 5, True),
+                             ('parallel', 'id', 3, 5, False),
+                             ('parallel', 'id', 3, 5, True),
+                             ('parallel', 'content_hash', 3, 5, False),
+                             ('parallel', 'content_hash', 3, 5, True)
+                         ])
+def test_cache_crud(
+        tmp_path,
+        indexers,
+        field,
+        shards,
+        chunks,
+        same_content
+):
+    config(field, tmp_path, shards, indexers)
+    print(f'{tmp_path=}')
+    f = Flow.load_config('yml/flow.yml')
+
+    docs = list(get_documents(chunks=chunks, same_content=same_content))
+
+    # initial data index
+    with f:
+        f.index(docs)
+
+    check_indexers_size(chunks, len(docs), field, tmp_path, same_content, shards, 'index')
+
+    # new documents
+    chunks_ids = np.concatenate([d.chunks for d in docs])
+    index_start_new_docs = 1 + max([int(d.id) for d in np.concatenate([chunks_ids, docs])])
+
+    new_docs = list(get_documents(chunks=chunks, same_content=same_content, index_start=index_start_new_docs))
+    with f:
+        f.index(new_docs)
+
+    check_indexers_size(chunks, len(docs), field, tmp_path, same_content, shards, 'index2')
+
+    # TODO update
+
+    docs.extend(new_docs)
+    # delete
+    with f:
+        f.delete(docs)
+
+    check_indexers_size(chunks, 0, field, tmp_path, same_content, shards, 'delete')
+
+
+def check_indexers_size(chunks, nr_docs, field, tmp_path, same_content, shards, post_op):
```

Comment:
This part is really too complex, and the code seems to assume a lot of features from a single test, which may make it hard to mantain. Every small change to things in the `indexers` can make this example fail.

Reply:
I think if we load the Indexer from the `indexer` yaml with the same workspace it will be able to load it from the `.bin` without this hardcode tricks

Reply:
This test should be self-contained. Any other special corner-cases should become their own tests.

Reply:
If you use `BaseExecutor.load_config` from the `indexer` uses you will be able to check the `size`. The test code would look much simpler